lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -251,14 +252,19 @@ define((require, exports, module) ->
       return token1.type == token2.type and token1.value == token2.value
 
     getEquationStart: (tokenIterator) ->
-      # if tokenIterator initialy is on the empty line, it's current token is null
+      # if tokenIterator is initialy on the empty line, its current token is null
       if not tokenIterator.getCurrentToken()?
         tokenIterator.stepForward()
-      # following cycle pushes tokenIterator to the end of
-      # beginning sequence, if it is inside one
-      for token in EquationRangeHandler.BEGIN_EQUATION_TOKEN_SEQUENCE
-        if EquationRangeHandler.equalTokens(token, tokenIterator.getCurrentToken())
-          tokenIterator.stepForward()
+        # if tokenIterator.getCurrentToken() is still null, then we're at the end of a file
+        if not tokenIterator.getCurrentToken()?
+          return null
+      else
```

Comment:
If the loop is run unconditionally, then there are problems with `\begin{equation}` positioned at the very beginning of a file or `\end{equation}` positioned at the very end of a file. 

Say, we position token iterator initially after the end of a file. First, we're looking for an end. We see, that current token is null, and move back one step. Then it is in the document already, and inside `\end{equation}` token sequence. Then the iterator is pushed to its start by a for loop, and then equation ending is found without any problems. But we should return null.